Dockerfile?

Is there a Dockerfile for building owt-server? I'm getting a few compile errors and using Docker as a standardized way to build a binary would be nice.

I started writing one.. and happy to share it.. But its getting stuck running the scripts/build.js building the webrtc code.. I think it is trying to build licode, but its missing the r_types.h header file.

Which ubuntu is best? 

Is there a demo server somewhere? 

FYI, I'm mostly interested in IP Camera RTSP broadcasting to 0 to N viewers. Saw your server supports RTSP, but not sure if it supports rtsp feeds as a webrtc source. 

Thanks!

I posted my first draft of a Dockerfile here:
https://github.com/Bhlowe/owt-server/blob/master/Dockerfile

If you run it using: 
`docker build -t owt-server . `

It fails building the scripts/build.js making webrtc. 

```
../WebRtcConnection.cc:543:16: error: 'using element_type = class erizo::WebRtcConnection {aka class erizo::WebRtcConnection}' has no member named 'removeRemoteCandidate'; did you mean 'addRemoteCandidate'?
   bool r = me->removeRemoteCandidate(mid, sdpMLine, sdp);
...
../WebRtcConnection.cc:361:17: error: 'class erizo::IceConfig' has no member named 'ip_addresses'
       iceConfig.ip_addresses = ipAddresses;
```

Any fix suggestions appreciated!

Also, anyone have a demo of this software running they can share that features live video?

> I posted my first draft of a Dockerfile here:
> https://github.com/Bhlowe/owt-server/blob/master/Dockerfile
> 
> If you run it using:
> `docker build -t owt-server . `
> 
> It fails building the scripts/build.js making webrtc.
> 
> ```
> ../WebRtcConnection.cc:543:16: error: 'using element_type = class erizo::WebRtcConnection {aka class erizo::WebRtcConnection}' has no member named 'removeRemoteCandidate'; did you mean 'addRemoteCandidate'?
>    bool r = me->removeRemoteCandidate(mid, sdpMLine, sdp);
> ...
> ../WebRtcConnection.cc:361:17: error: 'class erizo::IceConfig' has no member named 'ip_addresses'
>        iceConfig.ip_addresses = ipAddresses;
> ```
> Any fix suggestions appreciated!
> 
> Also, anyone have a demo of this software running they can share that features live video?

that means your licode source hasn't been patched. your git name and email should be set first. see installDeps.sh

That helped!
I updated my dockerfile.. Everything builds, but running the service doesn't seem to be working.
https://raw.githubusercontent.com/Bhlowe/owt-server/master/Dockerfile

```
wget https://raw.githubusercontent.com/Bhlowe/owt-server/master/Dockerfile
docker build -t owt .
docker run -it owt
```

This drops into bash. From there I can run:
./bin/init-all.sh --deps
./bin/start-all.sh

Now i'm just seeing the same kind of error in both init-all and start-all that looks like:

```
starting management-console, stdout -> /opt/owt-server/dist/logs/management-console.stdout
Start management-console HTTP server
starting app, stdout -> /opt/owt-server/dist/logs/app.stdout
module.js:549
    throw err;
    ^
Error: Cannot find module '/opt/owt-server/dist/extras/basic_example/samplertcservice.js'
```

--- a/docker/Dockerfile
+++ b/docker/Dockerfile
@@ -1,0 +1,38 @@
+################################
+# OWT WebRTC server
+#
+# Base image Ubuntu 18.04
+
+FROM ubuntu:18.04
+
+ENV DEBIAN_FRONTEND=noninteractive
+ENV MEDIA_SDK_VER=18.4.0
+
+# Update the system
+RUN apt update
+
+# Install utils
+RUN apt install -y ca-certificates git lsb-release mongodb nodejs npm sudo wget
+
+RUN npm install -g grunt-cli node-gyp
+
+# Keep proxy environment for sudo
+RUN bash -c "echo \"Defaults env_keep = \\\"http_proxy https_proxy no_proxy \\\"\" >> /etc/sudoers"
+
+# Install owt-client
+RUN git clone https://github.com/open-webrtc-toolkit/owt-client-javascript.git
+RUN cd owt-client-javascript/scripts && npm install && grunt
+
+RUN wget https://github.com/Intel-Media-SDK/MediaSDK/releases/download/intel-mediasdk-$MEDIA_SDK_VER/MediaStack.tar.gz -P /tmp
+RUN cd /tmp && tar -zxf MediaStack.tar.gz && cd MediaStack && ./install_media.sh
+RUN rm /tmp/MediaStack.tar.gz && rm -rf /tmp/MediaStack
+ENV MFX_HOME=/opt/intel/mediasdk
+
+RUN git clone https://github.com/open-webrtc-toolkit/owt-server.git
+WORKDIR /owt-server
+
+# This is needed to patch licode
+RUN  git config --global user.email "you@example.com" && \
+  git config --global user.name "Your Name"
+
+RUN ./scripts/installDepsUnattended.sh

--- a/docker/conference/Dockerfile-conference
+++ b/docker/conference/Dockerfile-conference
@@ -1,0 +1,13 @@
+################################
+# OWT WebRTC Conference Sample
+
+FROM owt-server:latest
+
+ENV CLIENT_SAMPLE_PATH=/owt-client-javascript/dist/samples/conference
+
+WORKDIR /owt-server
+
+RUN ./scripts/build.js -t all --check
+RUN ./scripts/pack.js -t all --install-module --sample-path $CLIENT_SAMPLE_PATH .
+
+RUN ./dist/video_agent/install_openh264.sh

--- a/docker/conference/docker-compose.yml
+++ b/docker/conference/docker-compose.yml
@@ -1,0 +1,26 @@
+version: '2'
+services:
+
+  owt-server:
+    build:
+      context: ../
+      dockerfile: Dockerfile
+      args:
+        - http_proxy
+        - https_proxy
+    image: owt-server
+  owt-server-conference:
+    build:
+      context: .
+      dockerfile: Dockerfile-conference
+      args:
+        - http_proxy
+        - https_proxy
+    image: owt-server-conference
+    network_mode: host
+    ports:
+      - "3004:3004"
+    command: sh -c "service mongodb start && service rabbitmq-server start \
+                    && cd /owt-server/dist/bin && ./init-all.sh && ./start-all.sh && sleep infinity"
+    depends_on:
+      - owt-server

